src/openrct2/world/map_generator/SimplexNoise.h

```diff
@@ -9,14 +9,61 @@
 
 #pragma once
 
+#include "Noise.h"
+
 #include <cstdint>
 
+#ifdef __WARN_SUGGEST_FINAL_METHODS__
+    #pragma GCC diagnostic push
+    #pragma GCC diagnostic ignored "-Wsuggest-final-methods"
+    #pragma GCC diagnostic ignored "-Wsuggest-final-types"
+#endif
```

Comment:
Why do you need to ignore the warning if you could just add the final keyword to the classes?

Reply:
IIRC only mingw complained about it and there are other locations that disable that warning

Reply:
Well you did not mark the final class as final such as SimplexFbmNoise